tensorflow_serving/model_servers/http_rest_api_util.cc

```diff
@@ -33,6 +33,12 @@ void AddHeaders(std::vector<std::pair<string, string>>* headers) {
   headers->push_back({"Content-Type", "application/json"});
 }
 
+void AddCORSHeaders(std::vector<std::pair<string, string>>* headers) {
+  headers->push_back({"Access-Control-Allow-Origin", "*"});
+  headers->push_back({"Access-Control-Allow-Methods", "POST, GET, OPTIONS"});
+  headers->push_back({"Access-Control-Allow-Headers", "Content-Type"});
```

Comment:
We can start with this. 

How will OPTIONS preflight be handled with this change?

Reply:
Thanks for the review.

ah right. I did not actually handle the OPTIONS request.
I think I need to call
```
evhttp_set_allowed_methods(ev_http_,
                             EVHTTP_REQ_GET | EVHTTP_REQ_POST | EVHTTP_REQ_HEAD |
                             EVHTTP_REQ_PUT | EVHTTP_REQ_DELETE | EVHTTP_REQ_OPTIONS);
```
so that the HttpRestApiHandler can actually handle the OPTIONS preflight.
And I also need to add
```
if (http_method == "OPTIONS") {
   // status = ...
}
```
in HttpRestApiHandler::ProcessRequest

I am not sure if I am on the right track?

Reply:
Is the goal here to just allow cross-origin requests without any security policies?

If this is the case, then you should provide a default implementation for OPTIONS, assuming OPTIONS is not used for same-origin requests (i.e. a supported REST method).

Reply:
yeah. I'll do that.

Reply:
Alternatively we should add the CORS handling in net_http and expose an API for the model module to configure the list of methods, content-type allowed.   Feel free to file an issue for a follow-up change.

It's somewhat involved to identify a cross-origin request, and for this PR, you can just check if there is a single Origin header for the OPTIONS request.  (400 if there are multiple ones).

Reply:
I guess Access-Control-Allow-Origin: "*" is too dangerous.

Should I implement it like another flag to set the allowed single origin?

Reply:
i agree with @wenbozhu that we should add CORS handling to net_http. but we can do that as a subsequent change, and not block this change for now.